agent/build.gradle.kts

```diff
@@ -26,104 +26,30 @@ dependencies {
   upstreamAgent("io.opentelemetry.javaagent:opentelemetry-javaagent")
 }
 
+licenseReport {
+  configurations = arrayOf(
+    project.configurations.bootstrapLibs.name,
+    project.configurations.javaagentLibs.name
+    // No need to include the upstreamAgent, because it already has all dependent licenses packaged
+    // Those will be preserved in our JAR
+  )
+}
 
 tasks {
 
-  val cleanLicenses by registering(Delete::class) {
-    Files.list(rootDir.toPath().resolve("licenses"))
-      .filter { f -> Files.isDirectory(f) || f.fileName.startsWith("NOTICE_") }
-      .forEach { f -> delete(f) }
-  }
-  generateLicenseReport {
-    dependsOn(cleanLicenses)
-  }
 
   // We override the agent entrypoints defined in elastic-otel.agent-packaging-convention
   shadowJar {
-
-    dependsOn(named("updateLicensesAndNotice"))
-
     // include licenses and notices in jar
-    from(rootDir) {
+    dependsOn(fullLicenseReport)
+    from(fullLicenseReport.get().outputs.files.singleFile) {
```

Comment:
Usually passing a task as input for another task is enough for gradle to create an implicit dependency between them and also to know what outputs to consume. Some cases might need things to be done explicitly though, maybe this is one of them, but if it's not, I think it'd be enough to replace lines 44 and 45 with `from(fullLicenseReport)`.

Reply:
Also, it seems like this config is applied the same way in the `agentextension` build file, if the same steps are needed for all cases then it'd probably be nice to make this work in the conventions file.

Reply:
> I think it'd be enough to replace lines 44 and 45 with from(fullLicenseReport)

Good point, i wasn't aware that `from()` accepts tasks as input, thanks!

> Also, it seems like this config is applied the same way in the agentextension build file, if the same steps are needed for all cases then it'd probably be nice to do this work in the conventions file.

I left that intentionally in the individual sub-projects: In my opinion it is easier to follow when the content-packaging of the JAR is less spread over multiple files, especially in this case because it is now only two duplicated lines across the two files.